src/modules/ekf2/EKF2.cpp

```diff
@@ -729,9 +729,15 @@ void EKF2::Run()
 		if (_ekf.update()) {
 			perf_set_elapsed(_ecl_ekf_update_full_perf, hrt_elapsed_time(&ekf_update_start));
 
-			PublishLocalPosition(now);
-			PublishOdometry(now, imu_sample_new);
-			PublishGlobalPosition(now);
+			if (_ekf.output_predictor().aligned()) {
+				// publish output predictor output
+				PublishLocalPosition(now);
```

Comment:
These could all be published earlier now, but I'm going to make that change after consolidating the reset handling so it's synchronized with the actual published output.